### DISCUSSION TITLE
Auto-generate Shopping List from Meal Plan

### DISCUSSION THREAD

**Discussion by lenaxia** [mealie-recipes/mealie#1899] [Feature Request]
**Before submitting this feature request I have**
_COMPLETE AND FILL THESE OUT_
- [X] Reviewed the documentation and verified that my feature request isn't already a feature.
- [X] Reviewed the existing feature requests and my feature has not been requested.
- [X] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
Currently there is no way to generate shopping lists based on meals for a day or week despite this feature clearly being visible here https://hay-kot.github.io/mealie/documentation/users-groups/meal-planner/

If this feature was removed, I request for it to be added back in as it is incredibly useful. Likewise the ability to generate entire meal plans for single meals should be brought back too.

This issue previously existed here: https://github.com/hay-kot/mealie/issues/62
and 

**(Optional): Suggest A Solution**
* In the meal planner, days should be able have clearly delineated meals (Brekkie, Lunch, Din, Snacks, Other)
* A button should exist for the day and for the "next 7 days" (potentially in a "..." menu, number of days could be configurable in settings) to "Generate a shopping list" based on all the meals included. 
* This would create a shopping list with ingredients for all meals included in the time frame
* In case a dish is repeated multiple times, all shopping list items would be prefixed with "2x Beef Tacos: " in the case that tacos will be made twice. e.g. "2x Beef Tacos: 2lbs of 80% lean beef" if the recipie calls for 1 lb of beef for each meal. 
* At this point, the user can delete any ingredients they already have. 

**Additional Information**
_FILL THESE OUT_
- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature

**Comment by michael-genson** [comment]
Since quite a few new features have been added surrounding shopping lists lately, I just want to add to this:
- Since we support recipe scaling, if a recipe is on a meal plan twice, the recipe should be added to the shopping list twice (i.e. a scale of 2)
- Since we support adding a subset of ingredients (instead of adding all ingredients) this should (optionally) be accessible for each recipe to be added

Now that we have better recipe support with shopping lists this is probably a smaller lift. Pretty sure it's exclusively a frontend task, now

**Reply by chuckberrypi** [reply]
Hey, this seems to have fallen by the wayside, but I'd really love to see this feature. Mealie has really promised to make meal planning convenient enough for me to actually do it, and streamlining the process of making a shopping list for the entire week would be amazing.

**Reply by michael-genson** [reply]
Adding one day's meals to your shopping list has been available since late last year: https://github.com/mealie-recipes/mealie/pull/2653

I agree we should still implement a "add all meal plans in my current view to the shopping list" though so you don't have to add each individual day.

**Comment by Hydro8** [comment]
Any update on this feature ?

I'm currently using "recipe keeper" on iOs with this feature : 
- On the weekly view, there is a "Add to shopping list" button
- When click on it, it shows all the recipes for the week (group by day and by type : lunch, dinner...)
- So you select the recipes that you need to buy something. You click on ok.
- It shows now the list of ingredients group by recipe. If there are the same recipe twice, it display the ingredients twice.
- After it adds the ingredients on the grocery list (same ingredients are merge : 100g cheese twice = 200g cheese).

I guess it works because when you plan the meal, you set for how many people. So if I set 2 peoples on monday and 2 peoples on tuesday (for 4 peoples recipe), the import shows me  1/2 ingredients for the monday and 1/2 ingredients for the tuesday.

At the end, I have all the ingredients on my list.

I saw that Mealie support recipe's portion to list so I guess with a workflow like that, it can work out of the box.
